packages/@aws-cdk/aws-cloudfront/lib/function.ts

```diff
@@ -0,0 +1,109 @@
+import { IResource, Names, Resource } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { CfnFunction } from './cloudfront.generated';
+
+/**
+ * Represents the function's source code
+ */
+export abstract class FunctionCode {
+
+  /**
+   * Inline code for function
+   * @returns `InlineCode` with inline code.
+   * @param code The actual function code
+   */
+  public static fromInline(code: string): InlineCode {
+    return new InlineCode(code);
+  }
+
+  /**
+   * renders the function code
+   */
+  public abstract render(): string;
+}
+
+export class InlineCode extends FunctionCode {
+
+  constructor(private code: string) {
+    super();
+  }
+
+  public render(): string {
+    return this.code;
+  }
+}
+
+/**
+ * Represents a CloudFront Function
+ */
+export interface IFunction extends IResource {
+  /**
+   * The name of the function.
+   * @attribute
+   */
+  readonly functionName: string;
+  /**
+   * The ARN of the function.
+   * @attribute
+   */
+  readonly functionArn: string;
+}
+
+/**
+ * Properties for creating a CloudFront Function
+ */
+export interface FunctionProps {
+  /**
+   * A name to identify the function.
+   * @default - generated from the `id`
+   */
+  readonly name?: string;
+
+  /**
+   * A comment to describe the function.
+   * @default - no comment
```

Comment:
looks from the specs that `comment` is required after all?

(not in the console but in the CF docs: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cloudfront-function-functionconfig.html#cfn-cloudfront-function-functionconfig-comment)

Reply:
Given that, let's update the docs.

```suggestion
   * @default - same as `functionName`
```